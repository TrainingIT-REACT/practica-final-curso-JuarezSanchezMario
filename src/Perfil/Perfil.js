import React, { Component } from "react";
import { getAlbums, getSongs } from "../reducers/historico";
// Css
import "./Perfil.css";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";
import Badge from "react-bootstrap/Badge";
import ListGroup from "react-bootstrap/ListGroup";
import ListGroupItem from "react-bootstrap/ListGroupItem";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

import { getUser } from "../reducers/user";

class Perfil extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  componentDidMount() {}

  render() {
    if (!this.props.user.logged) {
      return (
        <Redirect to={{ pathname: "/session", state: { from: "perfil" } }} />
      );
    }
    return (
      <div className="Perfil">
        <Row>
          <Col>
            <Row>
              <Badge variant="secondary">Albums visitados recientemente</Badge>
              {this.props.historicoAlbums.length ? (
                <ListGroup>
                  {this.props.historicoAlbums.map(item => (
                    <ListGroupItem>{item.name} by {item.artist}</ListGroupItem>
                  ))}
                </ListGroup>
              ) : (
                "Aun no has visitado ninguno"
              )}
            </Row>
          </Col>
          <Col>
            <Badge variant="secondary">
              Canciones escuchadas recientemente
            </Badge>
            {this.props.historicoSong.length ? (
                <ListGroup>
                  {this.props.historicoSong.map(item => (
                    <ListGroupItem>{item.name}</ListGroupItem>
                  ))}
                </ListGroup>
              ) : (
                "Aun no has escuchado ninguna"
              )}
          </Col>
        </Row>
      </div>
    );
  }
}

export default connect(state => ({
  user: getUser(state),
  historicoAlbums: getAlbums(state),
  historicoSong: getSongs(state)
}))(Perfil);
