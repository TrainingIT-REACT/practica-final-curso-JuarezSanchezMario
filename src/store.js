import { createStore, applyMiddleware, combineReducers } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from 'redux-devtools-extension';


// Reducers
import albums from "./reducers/albums";
import historico from "./reducers/historico";
import songs from "./reducers/songs";
import user from "./reducers/user";

export const initialState = {
  albums: {
    albums: [
      {
        id: 1,
        name: 'Chip off the old block',
        artist: 'Blair',
        cover: '/images/cover.jpg'
      },
      {
        id: 2,
        name: 'Battle grounds',
        artist: 'Florian',
        cover: '/images/cover.jpg'
      },
      {
        id: 3,
        name: 'Decisions decisions',
        artist: 'Skylar',
        cover: '/images/cover.jpg'
      },
      {
        id: 4,
        name: 'Blank canvas',
        artist: 'Skylar',
        cover: '/images/cover.jpg'
      },
      {
        id: 5,
        name: 'Ice cold',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      },
      {
        id: 6,
        name: 'Honesty',
        artist: 'Ted Garrett',
        cover: '/images/cover.jpg'
      },
      {
        id: 7,
        name: 'Creative director',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 8,
        name: 'No guarantees',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      },
      {
        id: 9,
        name: 'Don\'t push this button',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 10,
        name: 'Cherry',
        artist: 'Stacy',
        cover: '/images/cover.jpg'
      }
    ],
    loading: false,
    recomendados: [
      {
        id: 8,
        name: 'No guarantees',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      },
      {
        id: 9,
        name: 'Don\'t push this button',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 1,
        name: 'Chip off the old block',
        artist: 'Blair',
        cover: '/images/cover.jpg'
      },
      {
        id: 7,
        name: 'Creative director',
        artist: 'Delaney',
        cover: '/images/cover.jpg'
      },
      {
        id: 2,
        name: 'Battle grounds',
        artist: 'Florian',
        cover: '/images/cover.jpg'
      }
    ],
    selectedAlbum: {
      id: 5,
      name: 'Ice cold',
      artist: 'Kimberley',
      cover: '/images/cover.jpg'
    }
  },
  historico: {
    albums: [
      {
        id: 1,
        name: 'Chip off the old block',
        artist: 'Blair',
        cover: '/images/cover.jpg'
      },
      {
        id: 5,
        name: 'Ice cold',
        artist: 'Kimberley',
        cover: '/images/cover.jpg'
      }
    ],
    songs: [
      {
        id: 1,
        name: 'Take On Me',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 1
      },
      {
        id: 2,
        name: 'S.O.S.',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 1
      }
    ]
  },
  songs: {
    loading: false,
    songs: [
      {
        id: 21,
        name: 'Back In Black',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 22,
        name: 'Rock and Roll Ain\'t Noise Pollution',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 23,
        name: 'Dirty Deeds Done Dirt Cheap',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 24,
        name: 'Whole Lotta Rosie',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      },
      {
        id: 25,
        name: 'Let There Be Rock',
        audio: '/music/funky_energy_loop.mp3',
        seconds: 203,
        album_id: 5
      }
    ]
  },
  user: {
    name: 'Mario',
    email: 'mjs.mariojuarezsanchez@gmail.com',
    logged: true
  }
}

export default createStore(
  combineReducers({albums, historico, songs, user}),
  composeWithDevTools(applyMiddleware(thunk))
);
