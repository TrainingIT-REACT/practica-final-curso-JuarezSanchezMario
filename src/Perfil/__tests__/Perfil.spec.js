import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { initialState } from "../../store";
import thunk from "redux-thunk";


// Componente
import Perfil from '../Perfil';

describe("Perfil", () => {
  const mockStore = configureStore([thunk]);
  let store, wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Perfil />
      </Provider>
    );
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});