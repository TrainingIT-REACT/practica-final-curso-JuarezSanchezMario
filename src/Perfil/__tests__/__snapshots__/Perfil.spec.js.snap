// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Perfil renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Perfil)>
    <Perfil
      dispatch={[Function]}
      historicoAlbums={
        Array [
          Object {
            "artist": "Blair",
            "cover": "/images/cover.jpg",
            "id": 1,
            "name": "Chip off the old block",
          },
          Object {
            "artist": "Kimberley",
            "cover": "/images/cover.jpg",
            "id": 5,
            "name": "Ice cold",
          },
        ]
      }
      historicoSong={
        Array [
          Object {
            "album_id": 1,
            "audio": "/music/funky_energy_loop.mp3",
            "id": 1,
            "name": "Take On Me",
            "seconds": 203,
          },
          Object {
            "album_id": 1,
            "audio": "/music/funky_energy_loop.mp3",
            "id": 2,
            "name": "S.O.S.",
            "seconds": 203,
          },
        ]
      }
      user={
        Object {
          "email": "mjs.mariojuarezsanchez@gmail.com",
          "logged": true,
          "name": "Mario",
        }
      }
    >
      <div
        className="Perfil"
      >
        <Row
          noGutters={false}
        >
          <div
            className="row"
          >
            <Col>
              <div
                className="col"
              >
                <Row
                  noGutters={false}
                >
                  <div
                    className="row"
                  >
                    <Badge
                      pill={false}
                      variant="secondary"
                    >
                      <span
                        className="badge badge-secondary"
                      >
                        Albums visitados recientemente
                      </span>
                    </Badge>
                    <ListGroup
                      horizontal={null}
                      variant={null}
                    >
                      <ForwardRef
                        as="div"
                        className="list-group"
                        onSelect={[Function]}
                      >
                        <div
                          className="list-group"
                          onKeyDown={[Function]}
                        >
                          <ListGroupItem
                            active={false}
                            disabled={false}
                            variant={null}
                          >
                            <ForwardRef
                              as="div"
                              className="list-group-item"
                              disabled={false}
                              eventKey={null}
                              onClick={[Function]}
                            >
                              <div
                                className="list-group-item"
                                data-rb-event-key={null}
                                disabled={false}
                                onClick={[Function]}
                              >
                                Chip off the old block
                                 by 
                                Blair
                              </div>
                            </ForwardRef>
                          </ListGroupItem>
                          <ListGroupItem
                            active={false}
                            disabled={false}
                            variant={null}
                          >
                            <ForwardRef
                              as="div"
                              className="list-group-item"
                              disabled={false}
                              eventKey={null}
                              onClick={[Function]}
                            >
                              <div
                                className="list-group-item"
                                data-rb-event-key={null}
                                disabled={false}
                                onClick={[Function]}
                              >
                                Ice cold
                                 by 
                                Kimberley
                              </div>
                            </ForwardRef>
                          </ListGroupItem>
                        </div>
                      </ForwardRef>
                    </ListGroup>
                  </div>
                </Row>
              </div>
            </Col>
            <Col>
              <div
                className="col"
              >
                <Badge
                  pill={false}
                  variant="secondary"
                >
                  <span
                    className="badge badge-secondary"
                  >
                    Canciones escuchadas recientemente
                  </span>
                </Badge>
                <ListGroup
                  horizontal={null}
                  variant={null}
                >
                  <ForwardRef
                    as="div"
                    className="list-group"
                    onSelect={[Function]}
                  >
                    <div
                      className="list-group"
                      onKeyDown={[Function]}
                    >
                      <ListGroupItem
                        active={false}
                        disabled={false}
                        variant={null}
                      >
                        <ForwardRef
                          as="div"
                          className="list-group-item"
                          disabled={false}
                          eventKey={null}
                          onClick={[Function]}
                        >
                          <div
                            className="list-group-item"
                            data-rb-event-key={null}
                            disabled={false}
                            onClick={[Function]}
                          >
                            Take On Me
                          </div>
                        </ForwardRef>
                      </ListGroupItem>
                      <ListGroupItem
                        active={false}
                        disabled={false}
                        variant={null}
                      >
                        <ForwardRef
                          as="div"
                          className="list-group-item"
                          disabled={false}
                          eventKey={null}
                          onClick={[Function]}
                        >
                          <div
                            className="list-group-item"
                            data-rb-event-key={null}
                            disabled={false}
                            onClick={[Function]}
                          >
                            S.O.S.
                          </div>
                        </ForwardRef>
                      </ListGroupItem>
                    </div>
                  </ForwardRef>
                </ListGroup>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Perfil>
  </Connect(Perfil)>
</Provider>
`;
